components/style/properties/properties.mako.rs

```diff
@@ -596,40 +596,74 @@ pub fn parse_property_declaration_list(context: &ParserContext, input: &mut Pars
 }
 
 
-/// Only keep the last declaration for any given property.
+/// Only keep the "winning" declaration for any given property, by importance then source order.
 /// The input and output are in source order
 fn deduplicate_property_declarations(declarations: Vec<(PropertyDeclaration, Importance)>)
                                      -> Vec<(PropertyDeclaration, Importance)> {
-    let mut deduplicated = vec![];
-    let mut seen = PropertyBitField::new();
-    let mut seen_custom = Vec::new();
-    for declaration in declarations.into_iter().rev() {
-        match declaration.0 {
+    let mut deduplicated = Vec::new();
+    let mut seen_normal = PropertyBitField::new();
+    let mut seen_important = PropertyBitField::new();
+    let mut seen_custom_normal = Vec::new();
+    let mut seen_custom_important = Vec::new();
+    for (declaration, importance) in declarations.into_iter().rev() {
+        match declaration {
             % for property in data.longhands:
                 PropertyDeclaration::${property.camel_case}(..) => {
                     % if not property.derived_from:
-                        if seen.get_${property.ident}() {
-                            continue
+                        if importance.important() {
+                            if seen_important.get_${property.ident}() {
+                                continue
+                            }
+                            if seen_normal.get_${property.ident}() {
+                                remove_one(&mut deduplicated, |d| {
```

Comment:
Hmmm... I was wondering how common was to arrive here, and what's the normal length of the `deduplicated` vector, but it seems in average this is rather uncommon, and also not a huge deal.